lib/database/txserializer.js

```diff
@@ -1,13 +1,16 @@
 var to = require("../utils/to");
 var utils = require("ethereumjs-util");
 var FakeTransaction = require("ethereumjs-tx/fake.js");
+var Transaction = require("ethereumjs-tx");
 
 module.exports = {
   encode: function(tx, done) {
     var encoded = tx.toJSON(true);
 
     encoded.from = to.hex(tx.from);
-    encoded.hash = to.hex(tx.hash());
+    encoded.hash = to.txHash(tx);
+
+    encoded._isRawTx = tx._isRawTx;
```

Comment:
It does wind up breaking things since toJSON is consumed elsewhere - at the very least I noticed it produces conflicting hashes.  In addition, this is the only location I need to 'pass along' the transaction type for context later.